gradle/plugins/common/src/main/kotlin/junitbuild.java-library-conventions.gradle.kts

```
@@ -277,6 +281,10 @@ tasks {
 	checkstyleMain {
 		config = resources.text.fromFile(checkstyle.configDirectory.file("checkstyleMain.xml"))
 	}
+	assemble {
+		dependsOn("rewriteDryRun")
+		mustRunAfter("rewriteDryRun")
+	}
```

Did you mean to add this as an additional check?

```suggestion
	check {
		dependsOn("rewriteDryRun")
	}
```

---

I mean enforcing what's there or removing it if it seems unused.  

Currently, this is a one-time recipe for the "big bang," yes, but humans don't tend to migrate in their manner, meaning you apply the fixes once, but the new code is still written old (outdated) fashion.

Unfortunately, I don't like it either—neither "ether hop" nor "top." Applying this everywhere or not at all is about the "broken window" effect.  

Handling it locally, as shown with `spot` and `rewrite`, is the KISS (Keep It Simple) solution. A more advanced approach would involve a Git bot making suggestions/diffs, as seen (of course) in the rewrites themselves. While nice, it really overloads PRs in my humble opinion (and in @gnodet's opinion as well).  

- https://github.com/openrewrite/rewrite-static-analysis/pull/601#discussion_r2160166901